pdr_backend/analytics/predictoor_dashboard/dash_components/callbacks.py

```diff
@@ -0,0 +1,177 @@
+        search_value_predictoors,
+        feeds_data,
+        predictoors_data,
+        lake_dir,
+    ):
+        feeds_addrs = []
+        predictoors_addrs = []
+        if (
+            len(feeds_table_selected_rows) == 0
+            or len(predictoors_table_selected_rows) == 0
+        ):
+            return dash.no_update
+
+        current_feeds_table_data = update_feeds_table_on_search(
+            search_value_feeds, feeds_data
+        )
+        for i in feeds_table_selected_rows:
+            feeds_addrs.append(current_feeds_table_data[i]["contract"])
+
+        current_predictoors_table_data = update_predictoors_table_on_search(
+            search_value_predictoors, predictoors_data
+        )
+        for i in predictoors_table_selected_rows:
+            predictoors_addrs.append(current_predictoors_table_data[i]["user"])
+
+        payouts = get_payouts_from_db(feeds_addrs, predictoors_addrs, lake_dir)
+
+        return payouts
+
+    @app.callback(
+        Output("feeds_table", "columns"),
+        Input("feeds-data", "data"),
+    )
+    def create_feeds_table(feeds_data):
+        if not feeds_data:
+            return dash.no_update
+        for feed in feeds_data:
+            del feed["contract"]
+        columns = [{"name": col, "id": col} for col in feeds_data[0].keys()]
+        return columns
+
+    @app.callback(
+        Output("predictoors_table", "columns"),
+        Input("predictoors-data", "data"),
+    )
+    def create_predictoors_table(predictoors_data):
+        if not predictoors_data:
+            return dash.no_update
+        columns = [{"name": col, "id": col} for col in predictoors_data[0].keys()]
+        return columns
+
+    @app.callback(
+        Output("accuracy_chart", "children"),
+        Output("profit_chart", "children"),
+        Output("stake_chart", "children"),
+        Input("payouts-data", "data"),
+        Input("feeds_table", "selected_rows"),
+        Input("predictoors_table", "selected_rows"),
+        Input("search-input-Feeds", "value"),
+        Input("search-input-Predictoors", "value"),
+        State("feeds-data", "data"),
+        State("predictoors-data", "data"),
+    )
+    def create_charts(
+        payouts_data,
+        feeds_table_selected_rows,
+        predictoors_table_selected_rows,
+        search_value_feeds,
+        search_value_predictoors,
+        feeds_data,
+        predictoors_data,
+    ):
+        feeds_addrs = []
+        predictoors_addrs = []
+
+        ## calculate selected feeds
+        current_feeds_table_data = update_feeds_table_on_search(
+            search_value_feeds, feeds_data
+        )
+        for i in feeds_table_selected_rows:
+            feeds_addrs.append(
+                {
+                    "contract": current_feeds_table_data[i]["contract"],
+                    "feed_name": f"{current_feeds_table_data[i]['pair']}-{current_feeds_table_data[i]['timeframe']}",  # pylint: disable=line-too-long
```

Comment:
we have some objects like ArgPair that handle such cases. Could we maybe leverage those?